contracts/fuses/ramses/RamsesV2NewPositionFuse.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.26;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {PlasmaVaultConfigLib} from "../../libraries/PlasmaVaultConfigLib.sol";
+import {IFuseCommon} from "../IFuseCommon.sol";
+import {INonfungiblePositionManagerRamses} from "./ext/INonfungiblePositionManagerRamses.sol";
+import {FuseStorageLib} from "../../libraries/FuseStorageLib.sol";
+
+/// @notice Data for entering new position in Ramses V2
+struct RamsesV2NewPositionFuseEnterData {
+    /// @notice The address of the token0 for a specific pool
+    address token0;
+    /// @notice The address of the token1 for a specific pool
+    address token1;
+    /// @notice The fee associated with the pool Ramses V2 pool
+    uint24 fee;
+    /// @notice The lower end of the tick range for the position
+    int24 tickLower;
+    /// @notice The higher end of the tick range for the position
+    int24 tickUpper;
+    /// @notice The amount of token0 desired to be spent
+    uint256 amount0Desired;
+    /// @notice The amount of token1 desired to be spent
+    uint256 amount1Desired;
+    /// @notice The minimum amount of token0 to spend, which serves as a slippage check
+    uint256 amount0Min;
+    /// @notice The minimum amount of token1 to spend, which serves as a slippage check
+    uint256 amount1Min;
+    /// @notice Deadline for the transaction
+    uint256 deadline;
+    /// @notice The token ID of the RAM token
+    uint256 veRamTokenId;
+}
+
+/// @notice Data for closing position on Uniswap V3
+struct RamsesV2NewPositionFuseExitData {
+    /// @notice Token IDs to close, NTFs minted on Uniswap V3, which represent liquidity positions
+    uint256[] tokenIds;
+}
+
+/**
+ * @title RamsesV2NewPositionFuse
+ * @dev Contract for creating and managing new liquidity positions in the Ramses V2 system.
+ */
+contract RamsesV2NewPositionFuse is IFuseCommon {
+    using SafeERC20 for IERC20;
+
+    /// @notice Event emitted when a new position is created
+    /// @param version The address of the contract version
+    /// @param tokenId The ID of the token
+    /// @param liquidity The amount of liquidity added
+    /// @param amount0 The amount of token0 added
+    /// @param amount1 The amount of token1 added
+    /// @param token0 The address of token0
+    /// @param token1 The address of token1
+    /// @param fee The fee associated with the pool
+    /// @param tickLower The lower end of the tick range for the position
+    /// @param tickUpper The higher end of the tick range for the position
+    event RamsesV2NewPositionFuseEnter(
+        address version,
+        uint256 tokenId,
+        uint128 liquidity,
+        uint256 amount0,
+        uint256 amount1,
+        address token0,
+        address token1,
+        uint24 fee,
+        int24 tickLower,
+        int24 tickUpper
```

Comment:
There is no such data due to the error `CompilerError: Stack too deep. Try compiling with '--via-ir' (CLI) or the equivalent 'viaIR: true' (standard JSON) while enabling the optimizer. Otherwise, try removing local variables.` If necessary, we will be able to read it from the transaction or using the tokenId.